QuickOSM/ui/configuration_panel.py

```diff
@@ -24,28 +27,44 @@ class ConfigurationPanel(BasePanel):
 
     """Final implementation for the panel."""
 
-    def __init__(self, dialog):
+    def __init__(self, dialog: QDialog):
         super().__init__(dialog)
         self.panel = Panels.Configuration
 
     def setup_panel(self):
         """Set UI related the configuration panel."""
-        self.dialog.save_config.clicked.connect(self.set_server_overpass_api)
+        self.dialog.save_config_overpass.clicked.connect(self.set_server_overpass_api)
+        self.dialog.save_config_nominatim.clicked.connect(self.set_server_nominatim_api)
 
         for server in OVERPASS_SERVERS:
             self.dialog.combo_default_overpass.addItem(server)
 
+        for server in NOMINATIM_SERVERS:
+            self.dialog.combo_default_nominatim.addItem(server)
+
         # Read the config file
         custom_config = custom_config_file()
         if custom_config:
             with open(custom_config, encoding='utf8') as f:
                 config_json = load(f)
-                for server in config_json.get('overpass_servers'):
-                    if server not in OVERPASS_SERVERS:
-                        LOGGER.info(
-                            'Custom overpass server list added: {}'.format(
-                                server))
-                        self.dialog.combo_default_overpass.addItem(server)
+                try:
+                    for server in config_json.get('overpass_servers'):
```

Comment:
```suggestion
                overpass = config_json.get('overpass_server')
                for server in overpass:
```

Using `try`, `finally` is fragile and the code more difficult to read.